Classes/TYPO3/TYPO3CR/Search/Indexer/AbstractNodeIndexer.php

```diff
@@ -86,10 +86,41 @@ protected function evaluateEelExpression($expression, Node $node, $propertyName,
 	 * @param Node $node
 	 * @param string $propertyName
 	 * @param string $fulltextExtractionExpression
+	 * @param array array<TYPO3\Media\Domain\Model\AssetInterface>
+	 * @throws IndexingException
+	 * @return array
+	 */
+	protected function extractAssetsToFulltext(Node $node, $propertyName, $fulltextExtractionExpression, array &$fulltextIndexOfNode) {
+
+		if ($fulltextExtractionExpression !== '') {
+
+			$extractedFulltext = $this->evaluateEelExpression($fulltextExtractionExpression, $node, $propertyName, ($node->hasProperty($propertyName) ? $node->getProperty($propertyName) : NULL));
+
+			if (!is_array($extractedFulltext)) {
+				throw new IndexingException('The fulltext index for property "' . $propertyName . '" of node "' . $node->getPath() . '" could not be retrieved; the Eel expression "' . $fulltextExtractionExpression . '" is no valid fulltext extraction expression.');
+			}
+
+			foreach ($extractedFulltext as $bucket => $value) {
+				if (!isset($fulltextIndexOfNode[$bucket])) {
+					$fulltextIndexOfNode[$bucket] = array();
+				}
+				$fulltextIndexOfNode[$bucket] = $value ;
+			}
+		}
+
+		return $fulltextIndexOfNode;
+	}
```

Comment:
I am wondering whether we should really extract the assets to the "fulltext" of the index, or just add a separate property? I am currently totally unsure on that. Do you have use-cases why Fulltext Indexing makes sense? (I feel like they exist, but I do not see them yet).

Greets, Sebastian